### PR TITLE
#28: Explain relationship between host authentication and server certificate

### LINKED ISSUE #28: Add/Mention how to configure the Hostname/IP-Address for server certificate
Currently the tutorial section about using own certificates does not mention the requirement and setup of the hostname/ip-address for the server certificate ([see also](https://support.dnsimple.com/articles/what-is-common-name/)).

### PR DESCRIPTION
**Pull request by redcatbear**
Closes #28.

### CODE REVIEW

File: tls-tutorial/doc/use_your_own_certificate.md [hunk 1]
```diff
@@ -55,6 +55,20 @@ To follow this tutorial, you need
...
+
+Once you access the server, the client will work down a list of matching criteria to determine whether the certificate matches the server (see [RFC6125, 6.3](https://www.rfc-editor.org/rfc/rfc6125.html#page-21)). The whole process is non-trivial, so we encourage you at this point to take a look into the specification. It is well-written and contains relevant examples.
+
+In a nutshell, the client holds client-local information about the server against data presented in the server certificate. If they match, the server and certificate are considered authentic.
+
+While the standard states that the so called 'Common Name' (CN) from the certificate's subject is only the last way to match the hostname, it is still the most widely used variant. So for the sake of simplicity we are going to use this mechanism in this tutorial too.
+
+For you that means you need to know the hostname before you create the certificate.
+
+If you have a host that is intended to be reachable under multiple hostnames, please read about the concept of the "Subject Alternative Name" (see [RFC5280, 4.2.1.6](https://www.rfc-editor.org/rfc/rfc5280.html#page-35)). SAN is outside the scope of this tutorial.
```
Comment (kaklakariada) [inline tls-tutorial/doc/use_your_own_certificate.md]:
```suggestion
If you have a host that is intended to be reachable under multiple hostnames, please read about the concept of the "Subject Alternative Name" (SAN, see [RFC5280, 4.2.1.6](https://www.rfc-editor.org/rfc/rfc5280.html#page-35)). SAN is outside the scope of this tutorial. 
```

File: tls-tutorial/doc/use_your_own_certificate.md [hunk 2]
```diff
@@ -71,8 +85,10 @@ Do this to create a CA certificate:
     In a real world scenario you would obviously choose a secure, non-guessable passphrase.
 2. Create a CA certificate
    ```shell
-    openssl req -x509 -new -nodes -key ca.key -sha256 -days 365 -out ca.crt -subj '/CN=TLS Tutorial CA/C=DE/L=Bavaria/O=Tutorial Organization'
+    CERT_HOSTNAME='<fully-qualified-hostname>'
```
Comment (kaklakariada) [inline tls-tutorial/doc/use_your_own_certificate.md]:
```suggestion
    CERT_HOSTNAME='host.example.com'
```

**Review by kaklakariada — CHANGES_REQUESTED** [review]

**Review by kaklakariada — CHANGES_REQUESTED** [review]
Also add changelog entry

**Comment by redcatbear** [comment]
Done.

**Review by kaklakariada — APPROVED** [review]

**Pull request merged**